File: jvm-client/src/test/java/com/tomaszezula/makker/client/jvm/java/CreateScenario.java
package com.tomaszezula.makker.client.jvm.java;

import java.nio.file.Path;
import java.util.Base64;

import static com.tomaszezula.makker.client.jvm.java.Config.*;

public class CreateScenario {
    public static void main(String[] args) {
        fromBlueprint();
        fromEncodedBlueprint();
        fromFile();
    }

    private static void fromBlueprint() {
        makeClient.createScenario(
                        55228,
                        22143,
                        blueprint
                )
                .onSuccess(System.out::println)
                .onFailure(Throwable::printStackTrace);
    }

    private static void fromEncodedBlueprint() {
        makeClient.createScenarioEncoded(
                        55228,
                        22143,
                        Base64.getEncoder().encodeToString(blueprint.getBytes())
                )
                .onSuccess(System.out::println)
                .onFailure(Throwable::printStackTrace);
    }

    private static void fromFile() {
        makeClient.createScenario(
                        55228,
                        22143,
                        Path.of(getResource("blueprint.json"))
                )
                .onSuccess(System.out::println)
                .onFailure(Throwable::printStackTrace);
    }
}

File: jvm-client/src/test/kotlin/com/tomaszezula/makker/client/jvm/examples/CreateScenario.kt
package com.tomaszezula.makker.client.jvm.examples

import com.tomaszezula.makker.client.jvm.examples.Config.makeClient
import com.tomaszezula.makker.client.jvm.examples.Config.blueprint
import com.tomaszezula.makker.common.model.Blueprint
import com.tomaszezula.makker.common.model.Scenario
import io.kotest.common.runBlocking
import java.nio.file.Path
import java.util.*

fun main() {
    runBlocking {
        fromBlueprint()
        fromEncodedBlueprint()
        fromFile()
    }
}

private suspend fun fromBlueprint() {
    makeClient.createScenario(
        Scenario.TeamId(55228),
        Scenario.FolderId(22143),
        Blueprint.Json(blueprint)
    ).onSuccess {
        println(it)
    }.onFailure {
        it.printStackTrace()
    }
}

private suspend fun fromEncodedBlueprint() {
    makeClient.createScenario(
        Scenario.TeamId(55228),
        Scenario.FolderId(22143),
        Blueprint.Json(
            Base64.getEncoder().encodeToString(blueprint.toByteArray()),
            encoded = true
        )
    ).onSuccess {
        println(it)
    }.onFailure {
        it.printStackTrace()
    }
}

private suspend fun fromFile() {
    makeClient.createScenario(
        Scenario.TeamId(55228),
        Scenario.FolderId(22143),
        Path.of(Config.getResource("blueprint.json").toURI())
    ).onSuccess {
        println(it)
    }.onFailure {
        it.printStackTrace()
    }
}
